src/jltypes.c

```diff
@@ -1921,6 +1935,7 @@ void jl_init_types(void) JL_GC_DISABLED
     jl_true  = jl_permbox8(jl_bool_type, 1);
 
     jl_abstractstring_type = jl_new_abstracttype((jl_value_t*)jl_symbol("AbstractString"), core, jl_any_type, jl_emptysvec);
+    jl_abstractstring_type->incomplete = 0;
```

Comment:
shouldn't `jl_new_abstracttype` be setting `incomplete = 0` always (since you can never complete an abstract type).